src/EditorFeatures/Core/FindUsages/FindUsagesContext.cs

```diff
@@ -27,6 +25,8 @@ protected FindUsagesContext()
 
         public virtual Task OnReferenceFoundAsync(SourceReferenceItem reference) => Task.CompletedTask;
 
+        public virtual Task OnExternalReferenceFoundAsync(ExternalReferenceItem reference) => Task.CompletedTask;
```

Comment:
Is there a reason to have this be a different method than OnReferenceFoundAsync? It looks like for definitions we just have a single method with the base type?

Reply:
they don't really share anything in common.  And the consumer wants to know about how they differ as it impacts how the find-refs table works.  i could have something hacky, like have the ReferenceItems know about the find-refs window so we could create the right rows from it... but i really don't like that.  this is just effectively a way of saying "here's a minimal, non-smart, row that other clients can push information into find-refs with".